SW_VegProd.c

```diff
@@ -841,3 +865,539 @@ void get_critical_rank(void){
 		 End of rank_SWPcrits
 	 ----------------------------------------------------------*/
 }
+
+/**
+ @brief Wrapper function for estimating natural vegetation. First, climate is calculated and averaged, then values are estimated
+
+ @param[in,out] vegProd Structure holding all values for vegetation cover of simulation
+ @param[in] startYear Starting year of the simulation
+ @param[in] endYear Ending year of the simulation
+ @param[in] veg_method User specified value determining method of vegetation estimation with the current option(s):
+ 1 - Estimate fixed vegetation composition (fractional cover) from long-term climate conditions
+ */
+
+void estimateVegetationFromClimate(SW_VEGPROD *vegProd, int startYear, int endYear,
+                                   int veg_method, double latitude) {
+
+    int numYears = endYear - startYear + 1, deallocate = 0, allocate = 1, k;
+
+    SW_CLIMATE_YEARLY climateOutput;
+    SW_CLIMATE_CLIM climateAverages;
+
+    // NOTE: 8 = number of types, 5 = (number of types) - grasses
+
+    double coverValues[8] = {SW_MISSING, SW_MISSING, SW_MISSING, SW_MISSING,
+                            0.0, SW_MISSING, 0.0, 0.0}, shrubLimit = .2;
+
+    double SumGrassesFraction = SW_MISSING, C4Variables[3], grassOutput[3],
+    RelAbundanceL0[8], RelAbundanceL1[5];
+
+    Bool fillEmptyWithBareGround = swFALSE, warnExtrapolation = swTRUE;
+    Bool inNorthHem = swTRUE;
+
+    if(latitude < 0.0) {
+        inNorthHem = swFALSE;
+    }
+
+    // Allocate climate structs' memory
+    allocDeallocClimateStructs(allocate, numYears, &climateOutput, &climateAverages);
+
+    calcSiteClimate(SW_Weather.allHist, numYears, startYear, &climateOutput, inNorthHem);
+
+    averageClimateAcrossYears(&climateOutput, numYears, &climateAverages);
+
+    if(veg_method == 1) {
+
+        C4Variables[0] = climateAverages.minTemp7thMon_C;
+        C4Variables[1] = climateAverages.ddAbove65F_degday;
+        C4Variables[2] = climateAverages.frostFree_days;
+
+        estimatePotNatVegComposition(climateAverages.meanTemp_C, climateAverages.PPT_cm,
+                        climateAverages.meanTempMon_C, climateAverages.PPTMon_cm, coverValues,
+                        shrubLimit, SumGrassesFraction, C4Variables, fillEmptyWithBareGround,
+                        inNorthHem, warnExtrapolation, grassOutput, RelAbundanceL0, RelAbundanceL1);
+
+        ForEachVegType(k) {
+            vegProd->veg[k].cov.fCover = RelAbundanceL1[k];
+        }
+    }

```

Comment:
We also need to update fractional cover of bare ground (see "bare_cov.fCover" and last element of`RelAbundanceL1`) otherwise we could create fractional cover that sums to more than 100%